test/result_reporter_test.rb

```diff
@@ -0,0 +1,110 @@
+require "test_helper"
+
+class TestResult < MiniTest::Unit::TestCase
+
+  def test_build_results
+    result1 = RailsUpgradeAnalyzer::JobResult.new("1234.1", description: "Result 1", tests: 3, passed: 2, failures: 5, errors: 7)
+    result2 = RailsUpgradeAnalyzer::JobResult.new("1234.2", description: "Result 2", tests: 11, passed: 13, failures: 17, errors: 19)
+
+    reporter = RailsUpgradeAnalyzer::ResultReporter.new(result1, result2)
+
+    assert_equal "<table>", line(reporter, 0)
+    assert_equal "<thead>", line(reporter, 1)
+    assert_equal "<th>Branch</th>", line(reporter, 2)
+    assert_equal "<th>Tests</th>", line(reporter, 3)
+    assert_equal "<th>Passed</th>", line(reporter, 4)
+    assert_equal "<th>Failures</th>", line(reporter, 5)
+    assert_equal "<th>Errors</th>", line(reporter, 6)
+    assert_equal "<th>Passing %</th>", line(reporter, 7)
+    assert_equal "</thead>", line(reporter, 8)
+    assert_equal "<tbody>", line(reporter, 9)
+    assert_equal "<tr>", line(reporter, 10)
+    assert_equal "<td>Result 1</td>", line(reporter, 11)
+    assert_equal "<td>3</td>", line(reporter, 12)
+    assert_equal "<td>2</td>", line(reporter, 13)
+    assert_equal "<td>5</td>", line(reporter, 14)
+    assert_equal "<td>7</td>", line(reporter, 15)
+    assert_equal "<td>66.67</td>", line(reporter, 16)
+    assert_equal "</tr>", line(reporter, 17)
+
+    assert_equal "<tr>", line(reporter, 18)
+    assert_equal "<td>Result 2</td>", line(reporter, 19)
+    assert_equal "<td>11</td>", line(reporter, 20)
+    assert_equal "<td>13</td>", line(reporter, 21)
+    assert_equal "<td>17</td>", line(reporter, 22)
+    assert_equal "<td>19</td>", line(reporter, 23)
+    assert_equal "<td>118.18</td>", line(reporter, 24)
+    assert_equal "</tr>", line(reporter, 25)
+
+    assert_equal "</tbody>", line(reporter, 26)
+    assert_equal "</table>", line(reporter, 27)
```

Comment:
This took me a second to realize what was happening.  Almost feels like we should check equality against a docstring or array or something...

Just a bit surprising.  Most times, I wouldn't expect HTML to be checked this way.